chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -723,66 +723,53 @@ impl Ledger {
         Ok(())
     }
 
-    /// Try to apply messages to a State, and return the new State if successful
-    pub fn apply_block(
+    /// This function must be called before applying fragments to the ledger. Do not use it if you
```

Comment:
This seems like a good case for a wrapping type, like
```rust
pub struct LedgerApplyBlock(Ledger);

impl LedgerApplyBlock {
  pub fn apply_fragments(self, ...) -> LedgerApplyBlock { ... }
  pub fn finish(self, ...) -> Ledger { ... }
}
```
But considering that `apply_fragment` is already public and unrestricted, it doesn't add too much anyway... So I also wonder how useful is the comment here.

Reply:
I didn't notice @mzabaluev commented basically the same when I was writing this, so just answer to him (?)